java/build/src/main/java/sleeper/build/dependencydraw/DrawDependencyGraph.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.build.dependencydraw;
+
+import com.google.common.base.Function;
+import edu.uci.ics.jung.algorithms.layout.CircleLayout;
+import edu.uci.ics.jung.algorithms.layout.Layout;
+import edu.uci.ics.jung.graph.DirectedSparseGraph;
+import edu.uci.ics.jung.graph.Graph;
+import edu.uci.ics.jung.visualization.VisualizationViewer;
+import edu.uci.ics.jung.visualization.control.DefaultModalGraphMouse;
+import edu.uci.ics.jung.visualization.control.ModalGraphMouse;
+import edu.uci.ics.jung.visualization.decorators.ToStringLabeller;
+import edu.uci.ics.jung.visualization.renderers.Renderer;
+import org.jgrapht.alg.util.Pair;
+
+import sleeper.build.maven.DependencyReference;
+import sleeper.build.maven.MavenModuleAndPath;
+import sleeper.build.maven.MavenModuleStructure;
+
+import javax.swing.JCheckBox;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.SwingUtilities;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Paint;
+import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DrawDependencyGraph {
+    public boolean showTransitiveDependencies = false;
+
+    public Pair<List<String>, List<List<String>>> createGraph(List<MavenModuleAndPath> dependencies) {
+        List<String> nodeIds = new ArrayList<>();
+        List<List<String>> edges = new ArrayList<>();
+        for (MavenModuleAndPath maven : dependencies) {
+            nodeIds.add(maven.artifactReference().toString());
+            List<DependencyReference> nodeDependencies = maven.dependencies().
+                    filter(DependencyReference::isSleeper)
+                    .filter(DependencyReference::isExported)
+                    .collect(Collectors.toList());
+            for (DependencyReference dependency : nodeDependencies) {
+                List<String> listToAdd = new ArrayList<>();
+                listToAdd.add(maven.artifactReference().toString());
+                listToAdd.add(dependency.artifactReference().toString());
+                edges.add(listToAdd);
+            }
+        }
+        return new Pair<>(nodeIds, edges);
```

Comment:
It's hard to tell what this method is returning until you see this line. Instead of using a Pair, you could create a new class to represent a Graph, which will have nodeIds and edges as members, then you can use getters, and the return type of this function will make it clearer to see what's being returned.